persistence/share_repository.go

```diff
@@ -0,0 +1,54 @@
+package persistence
+
+import (
+	"context"
+	"time"
+
+	. "github.com/Masterminds/squirrel"
+	"github.com/astaxie/beego/orm"
+	"github.com/navidrome/navidrome/model"
+)
+
+type shareRepository struct {
+	sqlRepository
+}
+
+func NewShareRepository(ctx context.Context, o orm.Ormer) model.ShareRepository {
+	r := &shareRepository{}
+	r.ctx = ctx
+	r.ormer = o
+	r.tableName = "share"
+	return r
+}
+
+func (r *shareRepository) Delete(id string) error {
+	return r.delete(Eq{"id": id})
+}
+
+func (r *shareRepository) selectShare(options ...model.QueryOptions) SelectBuilder {
+	return r.newSelectWithAnnotation("share.id", options...).Columns("*")
```

Comment:
`r.newSelectWithAnnotation()` is only used by tables that can have annotations (star rating, love, play count, etc..) Tables that can be annotated are `album`, `artist`, `media_file` and `playlist`. This is not the case for the Share table, and even though it compiles I'm not even sure if this would work (tests would help here)

You should use `r.newSelect()` here.

Reply:
My bad! Thanks for the info. I've corrected it now.